tests/src/JIT/opt/CSE/NarrowStore.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace NarrowStore
+{
+    class Program 
+    { 
+        byte x01;
+        byte t01;
+    
+        int Test()
+        {
+            x01 = 3;
+            t01 = (byte)~x01;
```

Comment:
as I understand VN for t01((byte)~x01) was the same as VN for  (~x01) without byte cast, is it right?
Why do not we import this cast as explicit cast tree?

Reply:
I wouldn't really help, because the cast is only there because the C# compiler wants it.
In the orginal IL  repro icase there is just the assignment without any cast.

Reply:
THis isuse is that we record the value of t01 as ~3 or -4.  THen we optimzies the  next load to t01 as -4. Which we believeto be not equal to 252.  Instead we now perform the narrowing cast to unsigned byte on the -4 and know the value of t01 to be 252.

Reply:
Thanks.